lib/litmus.ex

```diff
@@ -1,18 +1,44 @@
 defmodule Litmus do
   @moduledoc """
-  Documentation for Litmus.
+  Documentation for Litmus. Validate data against schema defined.
   """
 
+  alias Litmus.Validation.{Params, Schema}
+
   @doc """
-  Hello world.
+  Validate your data against a schema
 
   ## Examples
 
-      iex> Litmus.hello
-      :world
+      iex(1)> alias Litmus.Type
+      Litmus.Type
+      iex(2)> login_schema = %{"id": %Type.Any{"required": true}, "username": %Type.String{"max_length": 20}}
+      %{
+        id: %Litmus.Type.Any{required: true},
+        username: %Litmus.Type.String{
+          length: nil,
+          max_length: 20,
+          min_length: nil,
+          regex: nil,
+          required: false,
+          trim: false
+        }
+      }
+      iex(3)> req_params = %{"id": 1, "username": "jane doe"}
+      %{id: 1, username: "jane doe"}
+      iex(4)> Litmus.validate(req_params, login_schema)
+      {:ok, %{id: 1, username: "jane doe"}}
```

Comment:
Might make sense to move the example up to the `@moduledoc` since there will really only be 1 function exposed to the user and we can show how the library looks higher up in the docs.